### PR TITLE
maxOfVector

### PR DESCRIPTION
**Pull request by ploWoj**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: max-of-vector
🏅 Granted 5 XP for @ploWoj

**Comment by ziobron** [comment]
⏱ 3 XP extra for @ploWoj for on-time delivery

File: homework/max-of-vector/maxOfVector.hpp [hunk 1]
```diff
@@ -1,8 +1,16 @@
 #pragma once
 #include <vector>
 #include <limits>
+#include <algorithm>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
-}
+    auto max = vec[0];
```
Comment (mariuszlisowski) [inline homework/max-of-vector/maxOfVector.hpp:7]: Dangerous! What if a vector is empty?

File: homework/max-of-vector/maxOfVector.hpp [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    for(const auto& el:vec)
+    {
+        if(max<el)
+        {
+            max = el;
+        }
+    }
+    return max;
+}
```
Comment (mariuszlisowski) [inline homework/max-of-vector/maxOfVector.hpp:16]: Newline needed at the end

File: homework/max-of-vector/maxOfVector.hpp [hunk 3]
```diff
@@ -1,8 +1,16 @@
 #pragma once
 #include <vector>
 #include <limits>
```
Comment (mariuszlisowski) [inline homework/max-of-vector/maxOfVector.hpp:3]: Included for a reason (probably). Hint: std::numeric_limits<>::

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]
Tests passed but would be good to predict passing an empty vector to a function thus changes are needed.

File: homework/max-of-vector/maxOfVector.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:7]: should be const

File: homework/max-of-vector/maxOfVector.hpp [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+#include <algorithm>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
-}
+    auto max = vec[0];
+    for(const auto& el:vec)
```
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:8]: pace between operations like '+' '-' '!=' '==' '%' etc...

File: homework/max-of-vector/maxOfVector.hpp [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+    {
+        if(max<el)
```
Comment (nauka-programowania-MA) [inline homework/max-of-vector/maxOfVector.hpp:10]: pace between operations like '+' '-' '!=' '==' '%' etc...

File: homework/max-of-vector/maxOfVector.hpp (hunk 2 above)
Comment (nauka-programowania-MA) [reply homework/max-of-vector/maxOfVector.hpp:16]: Actually we don't need it in modern compiler. This is legacy form C.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**